x/evm/keeper/abci.go

```diff
@@ -31,6 +31,13 @@ func (k *Keeper) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 	// reset counters that are used on CommitStateDB.Prepare
 	k.Bloom = big.NewInt(0)
 	k.TxCount = 0
+
+	//loading revision id from store in every block, make sure the `csdb.nextRevisionId` consistent in every node
+	id, err := k.GetRevisionID(ctx)
+	if err != nil {
+		panic(err)
+	}
+	k.CommitStateDB.SetRevisionID(id)
```

Comment:
I don't understand why you need to get the get the id from the store to then set it to the CSDB. Why not set it only during EndBlock using the KVStore that is already available on the CSDB?

Reply:
code removed